cwe_checker_rs/src/checkers/cwe_676.rs

```diff
@@ -0,0 +1,108 @@
+/// The CWE 676 checks for potentially dangerous function calls, such as strcpy, memcpy, scanf etc.
+
+use crate::{
+    intermediate_representation::{ Program, Project, Sub, Term, Tid, Jmp, ExternSymbol }, 
+    utils::log::{ CweWarning, LogMessage }};
+use serde::{ Serialize, Deserialize };
+
+const VERSION: &str = "0.1";
+
+pub static CWE_MODULE: crate::CweModule = crate::CweModule {
+    name: "CWE676",
+    version: VERSION,
+    run: check_cwe,
+};
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Hash, Clone)]
+pub struct Config {
+    symbols: Vec<String>,
+}
+
+
+/// For each block of a subroutine check if the jump is a direct call and if the target TID of that call is equal to a dangerous symbol TID
+pub fn get_call_to_target<'a>(caller: &'a Term<Sub>, target: &'a ExternSymbol) -> Vec<Option<(&'a str, &'a Tid, &'a str)>> {
+    let mut calls: Vec<Option<(&'a str, &'a Tid, &'a str)>> = Vec::new();
+    for blk in caller.term.blocks.iter() {
+        for jmp in blk.term.jmps.iter() {
+            match &jmp.term {
+                Jmp::Call{ target: dst, .. } => {
+                    if *dst == target.tid {
+                        calls.push(Some((caller.term.name.as_str(), &blk.tid, target.name.as_str())));
```

Comment:
You should use the `jmp.tid` instead of the `blk.tid` here. That we needed to use block TIDs was a restriction we do not have anymore.